wrappers/src/results_cs.cpp

```diff
@@ -118,14 +134,35 @@ REALM_EXPORT ManagedNotificationTokenContext* results_add_notification_callback(
         });
     });
 }
-
+    
 REALM_EXPORT Query* results_get_query(Results* results_ptr, NativeException::Marshallable& ex)
 {
     return handle_errors(ex, [&]() {
         return new Query(results_ptr->get_query());
     });
 }
     
+REALM_EXPORT Results* results_get_filtered_results(const Results& results, uint16_t* query_buf, size_t query_len, NativeException::Marshallable& ex)
+{
+    return handle_errors(ex, [&]() {
+        Utf16StringAccessor query_string(query_buf, query_len);
+        auto query = results.get_query();
+        auto const &realm = results.get_realm();
+        
+        parser::ParserResult result = parser::parse(query_string);
```

Comment:
Shouldn't we check whether the query was successfully parsed?

Reply:
I copied this verbatim from the js implementation and my assumption was that it would throw on error which would then get proxied by handle_errors.

Reply:
I can confirm it will throw runtime errors if something goes wrong.